Helper/Geolocation.php

```diff
@@ -0,0 +1,184 @@
+     * @param Context $context
+     * @param Config $configHelper
+     * @param Request $request
+     * @param Bugsnag $bugsnag
+     * @param ZendClientFactory $httpClientFactory
+     * @param CacheInterface $cache
+     *
+     * @codeCoverageIgnore
+     */
+    public function __construct(
+        Context $context,
+        ConfigHelper $configHelper,
+        Request $request,
+        Bugsnag $bugsnag,
+        ZendClientFactory $httpClientFactory,
+        CacheInterface $cache
+    ) {
+        parent::__construct($context);
+        $this->configHelper = $configHelper;
+        $this->request = $request;
+        $this->bugsnag = $bugsnag;
+        $this->httpClientFactory = $httpClientFactory;
+        $this->cache = $cache;
+    }
+
+    /**
+     * Gets the IP address of the requesting customer.
+     * This is used instead of simply $_SERVER['REMOTE_ADDR'] to give more accurate IPs if a proxy is being used.
+     *
+     * @return string  The IP address of the customer
+     */
+    private function getIpAddress() {
+
+        foreach (['HTTP_CLIENT_IP', 'HTTP_X_FORWARDED_FOR', 'HTTP_X_FORWARDED', 'HTTP_X_CLUSTER_CLIENT_IP',
+                     'HTTP_FORWARDED_FOR', 'HTTP_FORWARDED', 'REMOTE_ADDR',] as $key) {
+            if ($ips = $this->request->getServer($key, false)) {
+                foreach (explode(',', $ips) as $ip) {
+                    $ip = trim($ip); // just to be safe
+                    if (filter_var(
+                            $ip,
+                            FILTER_VALIDATE_IP,
+                            FILTER_FLAG_NO_PRIV_RANGE | FILTER_FLAG_NO_RES_RANGE
+                        ) !== false) {
+                        return $ip;
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @return string
+     */
+    private function getApiKey() {
+        return $this->configHelper->getGeolocationApiKey();
+    }
+
+    /**
+     * @param string $ip        The IP address
+     * @return null|string      JSON formated response
+     * @throws \Zend_Http_Client_Exception
+     */
+    private function getLocationJson($ip) {
+
+        $endpoint = sprintf($this->endpointFormat, $ip, $this->getApiKey());
+
+        $client = $this->httpClientFactory->create();
+        $client->setUri($endpoint);
+        $client->setConfig(['maxredirects' => 0, 'timeout' => 30]);
+
+        // Dependant on third party API, wrapping in try/catch block.
+        // On error notify bugsnag and proceed (return null)
+        try {
+            return $client->request()->getBody();
+        } catch (\Exception $e) {
+            $this->bugsnag->notifyException($e);
+            return null;
+        }
+    }
+
+    /**
+     * @param bool $useCache    flag that represents if Magento cache is used for reading / storing the result
+     * @return null|string      JSON formated response
+     * @throws \Zend_Http_Client_Exception
+     */
+    public function getLocation($useCache = true) {
```

Comment:
nit: currently we always set $useCache as true. Let's remove this param until we need it.

Reply:
done